src/hexkit/providers/akafka/provider/eventsub.py

```diff
@@ -181,76 +226,220 @@ async def construct(
+            error = ValueError(
+                f"Invalid value for retries_left: {retries_left} (should be between"
+                + f" 1 and {self._max_retries})"
+            )
+            logging.error(error)
+            raise error
+
+        # Decrement retries_left and calculate backoff time, then wait and retry.
+        retries_left -= 1
+        retry_number = self._max_retries - retries_left
+        backoff_time = self._retry_backoff * 2 ** (retry_number - 1)
+        try:
+            logging.info(
+                "Retry %i of %i for event of type '%s' on topic '%s' with key '%s',"
+                + " beginning in %i seconds.",
+                retry_number,
+                self._max_retries,
+                event.type_,
+                event.topic,
+                event.key,
+                backoff_time,
+            )
+            await asyncio.sleep(backoff_time)
+            await self._translator.consume(
+                payload=event.payload,
+                type_=event.type_,
+                topic=event.topic,
+                key=event.key,
+            )
+        except Exception as err:
+            if retries_left > 0:
+                await self._retry_event(event=event, retries_left=retries_left)
+            else:
+                raise self.RetriesExhaustedError(
+                    event_type=event.type_, max_retries=self._max_retries
+                ) from err
+
+    async def _handle_consumption(self, *, event: ExtractedEventInfo):
+        """Try to pass the event to the consumer.
+
+        If the event fails:
+        1. Retry until retries are exhausted, if retries are configured.
+        2. Publish the event to the DLQ topic if the DLQ is enabled. Done afterward.
+           or
+        3. Allow failure with unhandled error if DLQ is not configured.
+        """
+        try:
+            await self._translator.consume(
+                payload=event.payload,
+                type_=event.type_,
+                topic=event.topic,
+                key=event.key,
+            )
+        except Exception as underlying_error:
+            logging.warning(
+                "Failed initial attempt to consume event of type '%s' on topic '%s' with key '%s'.",
+                event.type_,
+                event.topic,
+                event.key,
+            )
+
+            if not self._max_retries:
+                if not self._enable_dlq:
+                    raise  # re-raise Exception
+                await self._publish_to_dlq(event=event)
+                return
+
+            # Don't raise RetriesExhaustedError unless retries are actually attempted
+            try:
+                await self._retry_event(event=event, retries_left=self._max_retries)
+            except (self.RetriesExhaustedError, ValueError) as retry_error:
+                # In case of ValueError, proceed with flow because that error is
+                # likely due to a hexkit bug, not a config/event error.
```

Comment:
This sounds a bit hacky. ValueError could also be produced by the translator or the core.

Reply:
If the ValueError was raised in the core or translator, it would have been caught in [in the retry method](https://github.com/ghga-de/hexkit/blob/b5ac74e1ace3cbb1414be72cea92bf41b06e1efc/src/hexkit/providers/akafka/provider/eventsub.py#L326) and ultimately re-raised as a RetriesExhaustedError. Maybe the comment should be removed or changed for clarity.

Reply:
See [27edd4a](https://github.com/ghga-de/hexkit/pull/123/commits/27edd4a45f1a0b25d7d66b8e8d008a1d576c7d1d)